src/index.js

```diff
@@ -10,33 +10,68 @@ function setOutput(label) {
 async function start() {
   const provider = config.input.backend
 
-  const start_instance_response = await slab.startInstanceRequest()
+  let start_instance_response
+
+  for (let i = 1; i <= 3; i++) {
+    try {
+      start_instance_response = await slab.startInstanceRequest()
+      break
+    } catch (error) {
+      core.info('Retrying request now...')
+    }
+
+    if (i === 3) {
+      core.setFailed(
+        `${provider} instance start request has failed after 3 attempts`
+      )
+    }
+  }
+
   core.info(
     `${provider} instance details: ${JSON.stringify(
       start_instance_response.details
     )}`
   )
 
-  const wait_instance_response = await slab.waitForInstance(
-    start_instance_response.task_id,
-    'start'
-  )
+  try {
+    const wait_instance_response = await slab.waitForInstance(
+      start_instance_response.task_id,
+      'start'
+    )
 
-  const instance_id = wait_instance_response.start.instance_id
-  core.info(`${provider} instance started with ID: ${instance_id}`)
+    const instance_id = wait_instance_response.start.instance_id
+    core.info(`${provider} instance started with ID: ${instance_id}`)
 
-  setOutput(start_instance_response.runner_name)
+    await waitForRunnerRegistered(start_instance_response.runner_name)
+  } catch (error) {
+    core.info(`Clean up after error, stop ${provider} instance`)
+    await slab.stopInstanceRequest(start_instance_response.runner_name)
```

Comment:
means the stop job will fail after I guess if the instance is already stopped ?

Reply:
The job will fail anyway since we set the action output at the end of the happy path.
It's debatable though. Do we let the user in charge of the teardown via a call to `stop` in his workflow ? But I'm not even sure we can get access to the output later if an action fails.

Reply:
could be a double security I guess, not sure, trying to still output something if the stop fails here, hoping the later stop manages to stop it in the end ?

Reply:
I move the output assignation at the earliest possible place so that a `stop` mode can try to terminate the instance even if `start` fails after the spawn request.